src/main/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImpl.java

```diff
@@ -429,38 +430,70 @@ protected String addACHAccount(Person payee, PayeeACHAccountExtractDetail achDet
         return processingError;
     }
 
-    /**
-     * Checks if the existing ACH account matches the data fed in from the input file, and creates and routes
-     * a new PAAT document to edit the account if necessary.
-     * 
-     * NOTE: The current implementation just logs match/mismatch status without making updates.
-     * This method will indicate that valid processing occurred whenever it is caled until it
-     * is modified in the future to perform actual updates.
-     */
     protected String updateACHAccountIfNecessary(Person payee, PayeeACHAccountExtractDetail achDetail, PayeeACHAccount achAccount) {
         StringBuilder processingResults = new StringBuilder();
-        processingResults.append("Update functionality NOT implemented yet. Payee \"")
-            .append(achDetail.getNetID()).append("\" data being checked for possible mismatch. ");
-
-        if (!StringUtils.equals(achDetail.getBankRoutingNumber(), achAccount.getBankRoutingNumber())
-                || !StringUtils.equals(achDetail.getBankAccountNumber(), achAccount.getBankAccountNumber())) {
-            /*
-             * For a future enhancement, we will modify this method to actually update the Payee ACH Account
-             * via a maintenance document, and possibly use more fields for change comparison.
-             */
-            processingResults.append(" Account information from input file for payee type '")
-                .append(achAccount.getPayeeIdentifierTypeCode())
-                .append("' does not match what is in KFS, but will be left as-is.");
+
+        if (accountHasChanged(achDetail, achAccount)) {
+            String accountUpdateErrors = updateACHAccount(payee, achDetail, achAccount);
+            processingResults.append(accountUpdateErrors);

```

Comment:
If updateACHAccount performs the update without errors, it return an empty string.  Would ti be worth checking for empty string, and adding a log message about a successful update?

Reply:
I think there's other spots that do this logging, but I'll see if I can clean up the logging to be clearer and more consistent.

Reply:
I've added and rearranged logging so that account changes (or lack of a need for changes) should now be logged appropriately.